**SOCIAL RESEARCH: Final Project** 

**NAME:** Letizia

**SURNAME:** Lanza

**ID:** VR502321

**PART 1:** Data Exploration and Data Wrangling

The first thing to succesfully carry out some interesting exercises, is to import some libraries as we have seen in class such as:

In [1]:
import os 
import numpy as np 
import pandas as pd 
import matplotlib.pyplot as plt 
import seaborn as sns

The aim of the following command is being able to read data from a CSV file and load it into a Pandas DataFrame.

In [2]:
obesity_df = pd.read_csv("ObesityDataSet_raw_and_data_sinthetic.csv")
obesity_df

Age  Gender    Height      Weight        CALC FAVC  FCVC  NCP  \
0     21.000000  Female  1.620000   64.000000          no   no   2.0  3.0   
1     21.000000  Female  1.520000   56.000000   Sometimes   no   3.0  3.0   
2     23.000000    Male  1.800000   77.000000  Frequently   no   2.0  3.0   
3     27.000000    Male  1.800000   87.000000  Frequently   no   3.0  3.0   
4     22.000000    Male  1.780000   89.800000   Sometimes   no   2.0  1.0   
...         ...     ...       ...         ...         ...  ...   ...  ...   
2106  20.976842  Female  1.710730  131.408528   Sometimes  yes   3.0  3.0   
2107  21.982942  Female  1.748584  133.742943   Sometimes  yes   3.0  3.0   
2108  22.524036  Female  1.752206  133.689352   Sometimes  yes   3.0  3.0   
2109  24.361936  Female  1.739450  133.346641   Sometimes  yes   3.0  3.0   
2110  23.664709  Female  1.738836  133.472641   Sometimes  yes   3.0  3.0   

      SCC SMOKE      CH2O family_history_with_overweight       FAF       TUE  \
0      no    no  2.000000                            yes  0.000000  1.000000   
1     yes   yes  3.000000                            yes  3.000000  0.000000   
2      no    no  2.000000                            yes  2.000000  1.000000   
3      no    no  2.000000                             no  2.000000  0.000000   
4      no    no  2.000000                             no  0.000000  0.000000   
...   ...   ...       ...                            ...       ...       ...   
2106   no    no  1.728139                            yes  1.676269  0.906247   
2107   no    no  2.005130                            yes  1.341390  0.599270   
2108   no    no  2.054193                            yes  1.414209  0.646288   
2109   no    no  2.852339                            yes  1.139107  0.586035   
2110   no    no  2.863513                            yes  1.026452  0.714137   

           CAEC                 MTRANS           NObeyesdad  
0     Sometimes  Public_Transportation        Normal_Weight  
1     Sometimes  Public_Transportation        Normal_Weight  
2     Sometimes  Public_Transportation        Normal_Weight  
3     Sometimes                Walking   Overweight_Level_I  
4     Sometimes  Public_Transportation  Overweight_Level_II  
...         ...                    ...                  ...  
2106  Sometimes  Public_Transportation     Obesity_Type_III  
2107  Sometimes  Public_Transportation     Obesity_Type_III  
2108  Sometimes  Public_Transportation     Obesity_Type_III  
2109  Sometimes  Public_Transportation     Obesity_Type_III  
2110  Sometimes  Public_Transportation     Obesity_Type_III  

[2111 rows x 17 columns]

As it can be seen the dataset has 2111 rows and 17 columns.

It is also important to explain what the 17 columns refer to: 
- Age: feature, continuous, "Age"
- Gender: feature, categorical, "Gender 
- Height: feature, continuous 
- Weight: feature, continuous 
- CALC: feature, categorical, "how often do you drink alcohol?"
- FAVC: feature, binary, "do you eat high caloric food frequently?"
- FCVC: feature, integer, "do you usually eat vegetables in your meals?"
- NCP: feature, continuous, "how many meals do you have daily?"
- SCC: feature, binary, "do you monitor the calories you eat daily?"
- SMOKE: feature, binary, "do you smoke?"
- CH2O: feature, continuous, "how much water do you drink daily?"
- family_history_with_overweight: feature, binary, "has a family member suffered or suffers from overweight?"
- FAF: feature, continuous, "how often do you have physical activity?"
- TUE: feature, integer, "how much time do you use technological devices such as cell phone, videogames, television, computer and others?"
- CAEC: feature, categorical, "do you eat any food between meals?"
- MTRANS: feature, categorical, "which transportation do you usually use?"
- NObeyesdad: target, categorical, "obesity level" 

More specifically, *"features"*, in a Social Research environment, are attributes or variables that are considered and analyzed to better understand social phenomena, relationships, and human behavior. Also, when a variable is *"continuous"* it means that it is a characteristic or attribute that can take on an infinite number of values within a specified range. These variables are measurable on a continuous scale and are not restricted to discrete or categorical values. In fact, when features are *"categorical"*, it signifies that they correspond variables that take discrete values and can be organized into distinct categories or levels rather than on a continuous scale. These categories can be represented by labels or nominal values that do not imply intrinsic ordering. Again, when features are *"binary"*, it indicates that they exemplify variables that take only two distinct values or categories. These two categories can be delineated by labels such as "0" and "1," "yes" and "no," "present" and "absent," "male" and "female," "success" and "failure," and so on. Finally, when features are of type *"integer"* it denotes that they represent numeric variables that take only integer values, that is, numbers without fractions or decimals. These variables are typically measurable on an integer scale and cannot take non - integer values.